### PR TITLE
Cap bal

### PR DESCRIPTION
**Pull request by toprince**
1. Add cap bal set and check when swap
2. add test case for set cap_bal

### CODE REVIEW

File: programs/woofi/src/errors.rs [hunk 1]
```diff
@@ -74,6 +74,8 @@ pub enum ErrorCode {
     NotEnoughOut, //0x178e
     #[msg("Amount out below minimum threshold")]
     AmountOutBelowMinimum, //0x178f
+    #[msg("Amount exceeds max balance cap")]
+    CapExceeds, //0x178f
```
Comment (fb-alexcq) [inline programs/woofi/src/errors.rs:78]: `BalanceCapExceeds`

Better off renaming as this.

File: programs/woofi/src/instructions/query.rs [hunk 2]
```diff
@@ -83,6 +88,11 @@ pub fn handler(ctx: Context<Query>, from_amount: u128, min_to_amount: u128) -> R
     let wooracle_to = &ctx.accounts.wooracle_to;
     let woopool_to = &ctx.accounts.woopool_to;
 
+    require!(
+        token_vault_from.amount as u128 + from_amount <= woopool_from.cap_bal,
```
Comment (fb-alexcq) [inline programs/woofi/src/instructions/query.rs:92]:
```
        (token_vault_from.amount as u128) + from_amount <= woopool_from.cap_bal
```

File: programs/woofi/src/instructions/swap.rs [hunk 3]
```diff
@@ -117,6 +117,11 @@ pub fn handler(ctx: Context<Swap>, from_amount: u128, min_to_amount: u128) -> Re
     let woopool_to = &mut ctx.accounts.woopool_to;
     let rebate_to = &ctx.accounts.rebate_to;
 
+    require!(
+        token_vault_from.amount as u128 + from_amount <= woopool_from.cap_bal,
```
Comment (fb-alexcq) [inline programs/woofi/src/instructions/swap.rs:121]: ditto

File: programs/woofi/src/errors.rs (hunk 1 above)
Comment (toprince) [reply programs/woofi/src/errors.rs:78]: done

File: programs/woofi/src/instructions/query.rs (hunk 2 above)
Comment (toprince) [reply programs/woofi/src/instructions/query.rs:92]: done

File: programs/woofi/src/instructions/swap.rs (hunk 3 above)
Comment (toprince) [reply programs/woofi/src/instructions/swap.rs:121]: done

**Pull request merged**